tests/integration/model-test.ts

```diff
@@ -680,4 +680,71 @@ module('Integration - Model', function (hooks) {
     assert.strictEqual(jupiter.name, 'Jupiter');
     assert.strictEqual(jupiter.hasName, true);
   });
+
+  test('hasOne contains type, id, and remoteId for loaded record', async function (assert) {
+    await store.addRecord<Planet>({
+      type: 'planet',
+      remoteId: '3',
+      name: 'Mars'
+    });
+
+    await store.source.update((t) => {
+      return t.addRecord({
+        type: 'moon',
+        id: '4-1',
+        attributes: {
+          name: 'Phobos'
+        },
+        relationships: {
+          planet: {
+            data: t.$normalizeRecordIdentity({
```

Comment:
I'm not 100% sure this correctly simulates what happens in a real app that's using a jsonapi source that gets back a server generated id here, but it does reproduce the same behavior I'm seeing in a real app.